lib/plugins/aws/deploy/lib/uploadArtifacts.js

```diff
@@ -1,21 +1,21 @@
 'use strict';
 
 const fs = require('fs');
-const path = require('path');
 const BbPromise = require('bluebird');
 const filesize = require('filesize');
+const path = require('path');
 
 module.exports = {
   uploadCloudFormationFile() {
     this.serverless.cli.log('Uploading CloudFormation file to S3...');
 
-    const body = JSON.stringify(this.serverless.service.provider.compiledCloudFormationTemplate);
```

Comment:
It would be important to keep the compiled template under the provider object. E.g. the alias plugin needs that information to create the alias stack information and modifies the compiled template. So it accesses this.serverless.service.provider.compiledCloudFormationTemplate to extract its information and modify the template in-memory. If the template is made local to a component, it will break all plugins that modify the template before it gets further processed by the AWSDeploy plugin.